samples/java/hibernate/README.md

```diff
@@ -79,4 +79,17 @@ public class User {
     @GeneratedValue
     private UUID id;
 }
+```
+
+## Optimization
+
+### Transaction
+Hibernate by default initiates every transaction as read-write even when there
+are only read queries. This causes over-head due to the locks required for read-write
+transaction.
+This can be optimized by initiating read only transaction for read queries.
+
+```java
+Session session = entityManager.unwrap(Session.class);
```

Comment:
This will make the entire session read-only. Would it be possible to also show an example for how you could do this for a single transaction?

Furthermore: We should add a test that verifies that this actually works as expected, and does start a read-only instead of a read/write transaction.

Reply:
Have removed this section as this was not working in testing. Will take is up as a separate bug.